In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
%matplotlib inline

### Problem Statement 1:

In each of the following situations, state whether it is a correctly stated hypothesis
testing problem and why?
1. H0: μ = 25, H1: μ ≠ 25
2. H0: σ > 10, H1: σ = 10
3. H0: x = 50, H1: x ≠ 50
4. H0: p = 0.1, H1: p = 0.5
5. H0: s = 30, H1: s > 30

1. 𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25

It is stated correctly because alternate hypothesis is totally opposite of null hypothesis and hypothesis always tested on population data.

2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10

It is stated correctly because it is tested on population standard deviation.

3. 𝐻0: x̅ = 50, 𝐻1: x̅ ≠ 50

It is not stated correctly because hypothesis testing done on population data. Not in sample data.(x̅ = sample mean)

4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5

It is not stated correctly because null and alternate hypothesis are opposit to each other. They never be the same.

5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30

It is stated incorrectly because hypothesis testing done on population data. Not in sample data. (s = sample statndard deviation)

### Problem Statement 2:

The college bookstore tells prospective students that the average cost of its textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics students thinks that the average cost is higher. To test the bookstore’s claim against their alternative, the students will select a random sample of size 100. Assume that the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the 5% level of significance and state your decision.

In [2]:
#H0 : mean of average cost of textbooks = 52

#H1 : mean of average cost of textbooks > 52

In [3]:
n = 100
mu = 52
sigma = 4.50
xbar = 52.80
alpha = 0.05
se = sigma/np.sqrt(n)  # Statndar Error
zstat = (xbar - mu)/se
zalpha = stats.norm.isf(0.05)
print('Z-Statistic value =',zstat)
print('Z-alpha value =',zalpha)
pvalue = stats.norm.sf(zstat)
print('P-value =',pvalue)

Z-Statistic value = 1.7777777777777715
Z-alpha value = 1.6448536269514729
P-value = 0.03772017981340073


As the Z-Statistic > Z-alpha and pvalue < 0.05 , we reject the null hypothesis and accept the alternate hypothesis i.e. mean of average cost of textbooks > 52

### Problem Statement 3:

A certain chemical pollutant in the Genesee River has been constant for several years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A group of factory representatives whose companies discharge liquids into the river is now claiming that they have lowered the average with improved filtration devices. A group of environmentalists will test to see if this is true at the 1% level of significance. Assume \ that their sample of size 50 gives a mean of 32.5 ppm. Perform a hypothesis test at the 1% level of significance and state your decision.

In [5]:
#H0 : mean of a certain chemical pollutant in the Genesee River = 34 ppm
#H1 : mean of a certain chemical pollutant in the Genesee River < 34 ppm

In [6]:
n = 50
mu = 34
sigma = 8
xbar = 32.5
alpha = 0.01
se = sigma/np.sqrt(n)  # Statndar Error
zstat = (xbar - mu)/se
zalpha = stats.norm.ppf(0.01)
print('Z-Statistic value =',zstat)
print('Z-alpha value =',zalpha)
pvalue = stats.norm.sf(zstat)
print('P-value =',pvalue)

Z-Statistic value = -1.3258252147247767
Z-alpha value = -2.3263478740408408
P-value = 0.9075512005172


As the mod(Z-Statistic) < mod(Z-alpha) and pvalue > 0.01 , we fail to reject the null hypothesis i.e. mean of a certain chemical pollutant in the Genesee River = 34 ppm

### Problem Statement 4:

Based on population figures and other general information on the U.S. population, suppose it has been estimated that, on average, a family of four in the U.S. spends about $1135 annually on dental expenditures. Suppose further that a regional dental association wants to test to determine if this figure is accurate for their area of country. To test this, 22 families of 4 are randomly selected from the population in that area of the country and a log is kept of the family’s dental expenditure for one year. The resulting data are given below. Assuming, that dental expenditure is normally distributed in the population, use the data and an alpha of 0.5 to test the dental association’s hypothesis.

1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994

In [9]:
#H0 : mean of dental expenditure of a family of 4 in a year = $1135

#H1 : mean of dental expenditure of a family of 4 in a year != $1135

In [10]:
n = 22
mu = 1135
sample = np.array([1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994])
xbar = sample.mean()
s = sample.std(ddof = 1)
alpha = 0.05
se = s/np.sqrt(n)   # Statndar Error
tstat = (xbar - mu)/se
talphaby2 = stats.t.ppf(alpha/2,n-1)
print('T-Statistic value =',tstat)
print('T-alpha value =',talphaby2)
pvalue = stats.t.cdf(tstat,n-1)*2
print('P-value =',pvalue)

T-Statistic value = -2.023137479931484
T-alpha value = -2.0796138447276626
P-value = 0.0559738319464585


As the mod(T-Statistic) < mod(T-alpha) and pvalue > 0.05 , we fail to reject the null hypothesis i.e. on an average, the dental expenditure of a family of 4 in a year = $1135

### Problem Statement 5:

In a report prepared by the Economic Research Department of a major bank the Department manager maintains that the average annual family income on Metropolis is 48,432. What do you conclude about the validity of the report if a random sample of 400 families shows and average income of 48,574 with a standard deviation of 2000?

In [12]:
#H0 : Average annual family income on Metropolis = 48,432
#H1 : Average annual family income on Metropolis != 48,432

In [13]:
n = 400
mu = 48432
xbar = 48574
s = 2000
alpha = 0.05
se = s/np.sqrt(n)# Statndar Error
tstat = (xbar - mu)/se
talphaby2 = stats.t.isf(alpha/2,n-1)
print('T-Statistic value =',tstat)
print('T-alpha value =',talphaby2)
pvalue = stats.t.cdf(tstat,n-1)*2
print('P-value =',pvalue)

T-Statistic value = 1.42
T-alpha value = 1.9659272959208094
P-value = 1.8436114582232594


As the mod(T-Statistic) < mod(T-alpha) and pvalue > 0.05 , we fail to reject the null hypothesis i.e. Average annual family income on Metropolis is 48,432

### Problem Statement 6:

Suppose that in past years the average price per square foot for warehouses in the United States has been 32.28. A national real estate investor wants to determine whether that figure has changed now. The investor hires a researcher who randomly samples 19 warehouses that are for sale across the United States and finds that the mean price per square foot is 31.67, with a standard deviation of 1.29. assume that the prices of warehouse footage are normally distributed in population. If the researcher uses a 5% level of significance, what statistical conclusion can be reached? What are the hypotheses?

In [14]:
#H0 : Average price per square foot for warehouses in the United States = 32.28
#H1 : Average price per square foot for warehouses in the United States != 32.28

In [15]:
n = 19
mu = 32.28
xbar = 31.67
s = 1.29
alpha = 0.05
se = s/np.sqrt(n)# Statndar Error
tstat = (xbar - mu)/se
talphaby2 = stats.t.ppf(alpha/2,n-1)
print('T-Statistic value =',tstat)
print('T-alpha value =',talphaby2)
pvalue = stats.t.cdf(tstat,n-1)*2
print('P-value =',pvalue)

T-Statistic value = -2.06118477175179
T-alpha value = -2.10092204024096
P-value = 0.0540335291599721


As the mod(T-Statistic) < mod(T-alpha) and pvalue > 0.05 , we fail to reject the null hypothesis i.e. Average price per square foot for warehouses in the United States is 32.28

### Problem Statement 7:

Fill in the blank spaces in the table and draw your conclusions from it.

In [16]:
n = 10
mu = 50
sigma = 2.5
ul = 51.5   # Upper Limit
ll = 48.5   # Lower Limit
se = sigma/np.sqrt(n)  # Statndar Error
zcrit = (ul - mu)/se   # Z-Critical value
alpha = stats.norm.sf(zcrit)*2  # The value of α
mu1 = 52
zstatll = (ll - mu1)/se  # for Z <= -4.43 
zstatul = (ul - mu1)/se  # for Z <= -0.63
beta = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll) # P(Z ≤ −0.63)−P(Z ≤ −4.43)
mu2 = 50.5
zstatll = (ll - mu2)/se  # Z <= -2.53 
zstatul = (ul - mu2)/se  # Z <= 1.27
beta2 = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll)  #= P(Z ≤ 1.27)−P(Z ≤ −2.53)
print ('The value of α in the acceptance area 48.5 < x̅ < 51.5 for sample size = 10 is',alpha)
print('The value of β at μ = 52 and acceptance area 48.5 < x̅ < 51.5 for sample size = 10 is',beta)
print('The value of β at μ = 50.5 and acceptance area 48.5 < x̅ < 51.5 for sample size = 10 is',beta2)

The value of α in the acceptance area 48.5 < x̅ < 51.5 for sample size = 10 is 0.057779571123597204
The value of β at μ = 52 and acceptance area 48.5 < x̅ < 51.5 for sample size = 10 is 0.26353985497284643
The value of β at μ = 50.5 and acceptance area 48.5 < x̅ < 51.5 for sample size = 10 is 0.8913423764409649


In [17]:
n = 10
mu = 50
sigma = 2.5
ul = 52
ll = 48
se = sigma/np.sqrt(n)
zstat = (ul - mu)/se
alpha = stats.norm.sf(zstat)*2  # The value of α
mu1 = 52
zstatll = (ll - mu1)/se 
zstatul = (ul - mu1)/se 
beta = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll)
mu2 = 50.5
zstatll = (ll - mu2)/se  
zstatul = (ul - mu2)/se  
beta2 = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll)
print ('The value of α in the acceptance area 48 < x̅ < 52 for sample size = 10 is',alpha)
print('The value of β at μ = 52 and acceptance area 48 < x̅ < 52 for sample size = 10 is',beta)
print('The value of β at μ = 50.5 and acceptance area 48 < x̅ < 52 for sample size = 10 is',beta2)

The value of α in the acceptance area 48 < x̅ < 52 for sample size = 10 is 0.011412036386001651
The value of β at μ = 52 and acceptance area 48 < x̅ < 52 for sample size = 10 is 0.4999997899803012
The value of β at μ = 50.5 and acceptance area 48 < x̅ < 52 for sample size = 10 is 0.9703275133092002


In [18]:
n = 16
mu = 50
sigma = 2.5
ul = 51.9
ll = 48.81
se = sigma/np.sqrt(n)
zstat = (ul - mu)/se
alpha = stats.norm.sf(zstat)*2  # The value of α
mu1 = 52
zstatll = (ll - mu1)/se 
zstatul = (ul - mu1)/se 
beta = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll)
mu2 = 50.5
zstatll = (ll - mu2)/se  
zstatul = (ul - mu2)/se  
beta2 = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll)
print ('The value of α in the acceptance area 48.81 < x̅ < 51.9 for sample size = 16 is',alpha)
print('The value of β at μ = 52 and acceptance area 48.81 < x̅ < 51.9 for sample size = 16 is',beta)
print('The value of β at μ = 50.5 and acceptance area 48.81 < x̅ < 51.9 for sample size = 16 is',beta2)

The value of α in the acceptance area 48.81 < x̅ < 51.9 for sample size = 16 is 0.00236578148620883
The value of β at μ = 52 and acceptance area 48.81 < x̅ < 51.9 for sample size = 16 is 0.4364403708344303
The value of β at μ = 50.5 and acceptance area 48.81 < x̅ < 51.9 for sample size = 16 is 0.9840290241059673


In [20]:
n = 16
mu = 50
sigma = 2.5
ul = 51.58
ll = 48.42
se = sigma/np.sqrt(n)
zstat = (ul - mu)/se
alpha = stats.norm.sf(zstat)*2  # The value of α
mu1 = 52
zstatll = (ll - mu1)/se 
zstatul = (ul - mu1)/se 
beta = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll)
mu2 = 50.5
zstatll = (ll - mu2)/se  
zstatul = (ul - mu2)/se  
beta2 = stats.norm.cdf(zstatul) - stats.norm.cdf(zstatll)
print ('The value of α in the acceptance area 48.42 < x̅ < 51.58 for sample size = 16 is',alpha)
print('The value of β at μ = 52 and acceptance area 48.42 < x̅ < 51.58 for sample size = 16 is',beta)
print('The value of β at μ = 50.5 and acceptance area 48.42 < x̅ < 51.58 for sample size = 16 is',beta2)

The value of α in the acceptance area 48.42 < x̅ < 51.58 for sample size = 16 is 0.011471435197072154
The value of β at μ = 52 and acceptance area 48.42 < x̅ < 51.58 for sample size = 16 is 0.25079184307889046
The value of β at μ = 50.5 and acceptance area 48.42 < x̅ < 51.58 for sample size = 16 is 0.9575685284788618


Conclusions:

The size of the critical region, and consequently the probability of a type I error α, can always be reduced by appropriate selection of the critical values.

• Type I and type II errors are related. When sample size n is fixed, a decrease in the probability of one type of error always results in an increase in the probability of the other.

• An increase in sample size will generally reduce both α and β, provided that the critical values are held constant.

• When the null hypothesis is false, β increases as the true value of the parameter approaches the value hypothesized in the null hypothesis

### Problem Statement 8:

Find the t-score for a sample size of 16 taken from a population with mean 10 when the sample mean is 12 and the sample standard deviation is 1.5.

In [23]:
n = 16
mu = 10
xbar = 12
s = 1.5
se = s/np.sqrt(n)
t_score = (xbar - mu)/se
print('t-score =',t_score)

t-score = 5.333333333333333


### Problem Statement 9:

Find the t-score below which we can expect 99% of sample means will fall if samples of size 16 are taken from a normally distributed population.

In [24]:
n = 16
cl = 0.99
alpha = 1 - cl
talpha = stats.t.ppf(alpha/2,n-1)
talpha

-2.946712883338615

t-score is in range -2.946712883338615 < t-score < 2.946712883338615

### Problem Statement 10:

If a random sample of size 25 drawn from a normal population gives a mean of 60 and a standard deviation of 4, find the range of t-scores where we can expect to find the middle 95% of all sample means. Compute the probability that (−𝑡0.05 <𝑡<𝑡0.10).

In [25]:
n = 25
cl1 = 0.95
alpha1 = 1 - cl1
alpha2 = 0.1
talpha1 = stats.t.ppf(alpha1,n-1)
talpha2 = stats.t.ppf(alpha2,n-1)
talpha1,talpha2

(-1.7108820799094275, -1.3178359318596027)

### Problem Statement 11:

Two-tailed test for difference between two population means Is there evidence to conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week, given the following:

Population 1: Bangalore to Chennai

n1 = 1200

x1 = 452

s1 = 212

Population 2: Bangalore to Hosur

n2 = 800

x2 = 523

s2 = 185

In [26]:
#H0 : population mean(Bangalore to Chennai) - population mean(Bangalore to Hosur) = 0

#H1 : population mean(Bangalore to Chennai) - population mean(Bangalore to Hosur) != 0

In [27]:
n1 = 1200
x1 = 452
s1 = 212
n2 = 800
x2 = 523
s2 = 185
alpha = 0.95
se = np.sqrt((s1**2/n1) + (s2**2/n2))
tstat = (x1 - x2)/se
talphaby2 = stats.t.ppf(alpha/2,n1+n2-2)
print('T-Statistic value =',tstat)
print('T-alpha value =',talphaby2)
pvalue = stats.t.cdf(tstat,n1+n2-2)*2
print('P-value =',pvalue)

T-Statistic value = -7.926428526759299
T-alpha value = -0.06271465548986041
P-value = 3.7154188748777314e-15


As the mod(T-Statistic) > mod(T-alpha) and pvalue < 0.05 , we reject the null hypothesis and accept the alternate hypothesis i.e. population mean(Bangalore to Chennai) - population mean(Bangalore to Hosur) != 0. So the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week

### Problem Statement 12:

Is there evidence to conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery, given the following:

Population 1: Duracell

n1 = 100

x1 = 308

s1 = 84

Population 2: Energizer

n2 = 100

x2 = 254

s2 = 67

In [28]:
#H0 : population mean(Duracell Battery) - population mean(Energizer Battery) = 0

#H1 : population mean(Duracell Battery) - population mean(Energizer Battery) > 0

In [29]:
n1 = 100
x1 = 308
s1 = 84
n2 = 100
x2 = 254
s2 = 67
alpha = 0.95
se = np.sqrt((s1**2/n1) + (s2**2/n2))
tstat = (x1 - x2)/se
talphaby2 = stats.t.ppf(alpha,n1+n2-2)
print('T-Statistic value =',tstat)
print('T-alpha value =',talphaby2)
pvalue = stats.t.sf(tstat,n1+n2-2)
print('P-value =',pvalue)

T-Statistic value = 5.025702668336442
T-alpha value = 1.6525857836172075
P-value = 5.592823870574184e-07


As the mod(T-Statistic) > mod(T-alpha) and pvalue < 0.05 , we reject the null hypothesis and accept the alternate hypothesis i.e. population mean(Duracell Battery) - population mean(Energizer Battery) > 0. So the number of people preferring Duracell battery is different from the number of people preferring Energizer battery.

### Problem Statement 13:

Pooled estimate of the population variance

Does the data provide sufficient evidence to conclude that average percentage increase in the price of sugar differs when it is sold at two different prices?

Population 1: Price of sugar = Rs. 27.50

n1 = 14

x1 = 0.317%

s1 = 0.12%

Population 2: Price of sugar = Rs. 20.00

n2 = 9

x2 = 0.21%

s2 = 0.11%

In [31]:
#H0 : population1 mean - population2 mean = 0

#H1 : population1 mean - population2 mean > 0

In [32]:
n1 = 14
x1 = 0.317
s1 = 0.12
n2 = 9
x2 = 0.21
s2 = 0.11
stats.ttest_rel([x1,x2],[n1,n2])

Ttest_relResult(statistic=-4.592887798896382, pvalue=0.1364798873881824)

As pvalue > 0.05 , we fail to reject the null hypothesis i.e. population1 mean - population2 mean = 0. So average percentage increase in the price of sugar does not differ when it is sold at two different prices.

### Problem Statement 14:

The manufacturers of compact disk players want to test whether a small price reduction is enough to increase sales of their product. Is there evidence that the small price reduction is enough to increase sales of compact disk players?

Population 1: Before reduction

n1 = 15

x1 = Rs. 6598

s1 = Rs. 844

Population 2: After reduction

n2 = 12

x2 = RS. 6870

s2 = Rs. 669

In [33]:
#H0 : population1 mean - population2 mean = 0

#H1 : population1 mean - population2 mean > 0

In [34]:
n1 = 15
x1 = 6598
s1 = 844
n2 = 12
x2 = 6870
s2 = 669
stats.ttest_rel([x1,x2],[n1,n2])

Ttest_relResult(statistic=48.876363636363635, pvalue=0.013023288088995388)

As pvalue < 0.05 , We reject the null hypothesis and accept the alternate hypothesis i.e. population1 mean - population2 mean > 0. So the small price reduction is enough to increase sales of compact disk players.

### Problem Statement 15:

Comparisons of two population proportions when the hypothesized difference is zero Carry out a two-tailed test of the equality of banks’ share of the car loan market in 1980 and 1995.

Population 1: 1980

n1 = 100

x1 = 53

𝑝1 = 0.53

Population 2: 1985

n2 = 100

x2 = 43

𝑝2= 0.43

In [36]:
#H0 : population1 mean - population2 mean = 0

#H1 : population1 mean - population2 mean != 0

In [37]:
n1 = 100
x1 = 53
p1 = 0.53
n2 = 100
x2 = 43
p2= 0.43
alpha = 0.95
p = (x1+x2)/(n1+n2)
q = 1 - p
z = (p1 - p2)/np.sqrt(p*q*((1/n1)+(1/n2)))
zalpha = stats.norm.ppf(alpha)
print('Z-Statistic value =',z)
print('Z-alpha value =',zalpha)

Z-Statistic value = 1.4153462926807452
Z-alpha value = 1.6448536269514722


As Z-Statistic < Z-Alpha , we fail to reject null hypothesis i.e. population1 mean - population2 mean = 0. So the equality of banks’ share of the car loan market in 1980 and 1995 is same.

### Problem Statement 16:

Carry out a one-tailed test to determine whether the population proportion of traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes are offered as at least 10% higher than the proportion of such buyers when no sweepstakes are on.

Population 1: With sweepstakes

n1 = 300

x1 = 120

𝑝1 = 0.40

Population 2: No sweepstakes

n2 = 700

x2 = 140

𝑝2= 0.20

In [38]:
#H0 : population1 mean - population2 mean <= 0.1

#H1 : population1 mean - population2 mean > 0.1

In [39]:
n1 = 300
x1 = 120
p1 = 0.40
n2 = 700
x2 = 140
p2= 0.20
alpha = 0.001
d = 0.1
q1 = 1 - p1
q2 = 1 - p2
z = ((p1 - p2) - d)/np.sqrt(((p1*q1)/n1)+((p2*q2)/n2))
zalpha = stats.norm.isf(alpha)
print('Z-Statistic value =',z)
print('Z-alpha value =',zalpha)

Z-Statistic value = 3.1180478223116177
Z-alpha value = 3.090232306167813


As Z-Statistic < Z-Alpha , we reject null hypothesis and accept alternate hypothesis i.e. population1 mean - population2 mean > 0.10. So the population proportion of traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes are offered as at least 10% higher than the proportion of such buyers when no sweepstakes are on.

### Problem Statement 17:

A die is thrown 132 times with the following results: Number turned up: 1, 2, 3, 4, 5, 6

Frequency: 16, 20, 25, 14, 29, 28

Is the die unbiased? Consider the degrees of freedom as 𝑝 − 1.

In [41]:
#H0 : The die is biased.

#H1 : The die is not biased.

In [42]:
observation = np.array([16, 20, 25, 14, 29, 28])
expected = np.ones(6)*22
stats.chisquare(observation,expected)

Power_divergenceResult(statistic=9.0, pvalue=0.1090641579497725)

As pvalue > 0.05 , we fail to reject the null hypothesis i.e. the die is biased.

### Problem Statement 18:

In a certain town, there are about one million eligible voters. A simple random sample of 10,000 eligible voters was chosen to study the relationship between gender and participation in the last election. The results are summarized in the following 2X2 (read two by two) contingency table:


We would want to check whether being a man or a woman (columns) is independent of having voted in the last election (rows). In other words, is “gender and voting independent”?

In [43]:
#H0 : gender and voting are independent

#H1 : gender and voting are not independent

In [44]:
n = np.array([2792,3591,1486,2131]).reshape(2,2)
n

array([[2792, 3591],
       [1486, 2131]])

In [45]:
stats.chi2_contingency(n)

(6.552337973968181,
 0.01047464310091866,
 1,
 array([[2730.6474, 3652.3526],
        [1547.3526, 2069.6474]]))

As P-value is less than 0.05 so we reject null hypothesis and accept the alternate hypothesis i.e. Gender and Voting are dependent.

### Problem Statement 19:

A sample of 100 voters are asked which of four candidates they would vote for in an election. The number supporting each candidate is given below:



Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96, with 3 df, 𝑝 < 0.05 ].

In [46]:
#H0 : all candidates are equally popular

#H1 : all candidates are not equally popular

In [47]:
n = np.array([41,19,24,16])
stats.chisquare(n)

Power_divergenceResult(statistic=14.959999999999999, pvalue=0.001851152735284203)

As P-value is less than 0.05 so we reject null hypothesis and accept the alternate hypothesis i.e. all candidates are not equally popular.

### Problem Statement 20:

Children of three ages are asked to indicate their preference for three photographs of adults. Do the data suggest that there is a significant relationship between age and photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4 df: 𝑝 < 0.05].

In [48]:
#Ho : there is a significant relationship between age and photograph preference

#H1 : there is no significant relationship between age and photograph preference

In [49]:
table = np.array([18,22,20,2,28,40,20,10,40]).reshape(3,3)
stats.chi2_contingency(table)

(29.603174603174608,
 5.894582174262606e-06,
 4,
 array([[12., 18., 30.],
        [14., 21., 35.],
        [14., 21., 35.]]))

As P-value is less than 0.05 so we reject null hypothesis and accept the alternate hypothesis i.e. there is no significant relationship between age and photograph preference.

### Problem Statement 21:

A study of conformity using the Asch paradigm involved two conditions: one where one confederate supported the true judgement and another where no confederate gave the correct response.


Is there a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df: 𝑝 < 0.05].

In [50]:
#H0 : there is no significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform

#H1 : there is a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform

In [51]:
table = np.array([18,40,32,10]).reshape(2,2)
stats.chi2_contingency(table)

(18.10344827586207,
 2.092221924065813e-05,
 1,
 array([[29., 29.],
        [21., 21.]]))

As P-value is less than 0.05 so we reject null hypothesis and accept the alternate hypothesis i.e. there is a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform.

### Problem Statement 22:

We want to test whether short people differ with respect to their leadership qualities (Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget MP's are there?) The following table shows the frequencies with which 43 short people and 52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a relationship between height and leadership qualities?

[Chi-Square = 10.71, with 2 df: 𝑝 < 0.01].

In [53]:
#H0 : there is a relationship between height and leadership qualities

#H1 : there is no relationship between height and leadership qualities

In [54]:
table = np.array([12,32,22,14,9,6]).reshape(3,2)
stats.chi2_contingency(table)

(10.712198008709638,
 0.004719280137040844,
 2,
 array([[19.91578947, 24.08421053],
        [16.29473684, 19.70526316],
        [ 6.78947368,  8.21052632]]))

As P-value is less than 0.01 so we reject null hypothesis and accept the alternate hypothesis i.e. there is no relationship between height and leadership qualities.

### Problem Statement 23:

Each respondent in the Current Population Survey of March 1993 was classified as employed, unemployed, or outside the labor force. The results for men in California age 35- 44 can be cross-tabulated by marital status, as follows:



Men of different marital status seem to have different distributions of labor force status. Or is this just chance variation? (you may assume the table results from a simple random sample.)

In [55]:
#H0 : There is a relationship between Marital Status and Employment

#H1 : There is no relationship between Marital Status and Employment

In [56]:
table = np.array([679,103,114,63,10,20,42,18,25]).reshape(3,3)
stats.chi2_contingency(table)

(31.61310319407798,
 2.2949883042859918e-06,
 4,
 array([[654.06331471, 109.2886406 , 132.64804469],
        [ 67.88826816,  11.34357542,  13.76815642],
        [ 62.04841713,  10.36778399,  12.58379888]]))

As P-value is less than 0.05 so we reject null hypothesis and accept the alternate hypothesis i.e. There is no relationship between Marital Status and Employment.